src/Libraries/CoreNodes/Data.cs

```diff
@@ -51,16 +70,114 @@ private static object ToNative(JToken token)
             }
         }
 
+        /// <summary>
+        /// Parse implementation for converting JObject types to specific Dynamo objects (ie Geometry, Color, Images, etc) 
+        /// </summary>
+        /// <param name="jObject"></param>
+        /// <returns></returns>
+        /// <exception cref="Exception"></exception>
+        private static object DynamoJObjectToNative(JObject jObject)
+        {
+            if (jObject.ContainsKey("$typeid"))
+            {
+                var typeid = jObject["$typeid"].ToString();
+
+                switch (typeid)
+                {
+                    //autodesk.math to abstract ProtoGeometry types
+                    case "autodesk.math:vector3d-1.0.0":
+                        return Vector.FromJson(jObject.ToString());
+
+                    case "autodesk.math:matrix44d-1.0.0":
+                        return CoordinateSystem.FromJson(jObject.ToString());
+
+                    //autodesk.geometry to abstract ProtoGeometry types
+                    case "autodesk.geometry:boundingbox3d-1.0.0":
+                        return BoundingBox.FromJson(jObject.ToString());
+
+                    case "dynamo.geometry:mesh-1.0.0":
```

Comment:
is this defined in the registry, or just in our code directly?

Reply:
Our code directly until we get it registered.  That process takes a bit.

Reply:
is there a task somewhere for it?

Reply:
Yes in the Vera backlog.  A few things we are following up on with the forge schemas.  1) Adding our Dynamo types where it makes senses. and 2) Adding support to the geometry schemas for things we need and make sense for a broader audience... for oriented bounding box, Rectangles, Cuboid Surfaces...  The mesh one is bit unclear.  There isn't an open source definition that anyone can agree on.  TrittonMesh is not open now.  We have a task to support glt as the standard mesh type but that is also complicated.  I think the main pattern is that we will 1) document these and 2) begin to introduce non dynamo specific schemas over time.  In those cases we will continue to deserialize these types but serialize the new types.